tests/test_user_login.py

```diff
@@ -0,0 +1,65 @@
+"""
+This file holds user login test cases.
+"""
+import unittest
+import json
+from api import create_app
+
+
+class AuthTestCase(unittest.TestCase):
+    """
+        Test case for User login.
+    """
+
+    def setUp(self):
+        """Set up test variables."""
+        self.app = create_app(config_name="testing")
+        # initialize the test client
+        self.client = self.app.test_client
+        # This is the user test json data with a predefined email and password
+        self.user_data = {
+            'user_email': 'jimmy@example.com',
+            'password': 'test_123'
+        }
+        self.user_data_log = {
+            'user_email': 'kennedy@example.com',
+            'password': 'test_123'
+        }
+        self.not_a_user = {
+            'user_email': 'abela@test.com',
+            'password': 'nopekabisa'
+        }
+
+    def test_user_login(self):
+        """Test signed up user can login."""
+        response = self.client().post('/api/v1/auth/signup', data=self.user_data_log)
+        self.assertEqual(response.status_code, 201)
+        login_response = self.client().post('/api/v1/auth/login', data=self.user_data_log)
+        #assert that the status code is equal to 200
+        self.assertEqual(login_response.status_code, 200)
+        #return result in json format
+        result = json.loads(login_response.data.decode())
+        #test that the response contains success message
+        self.assertEqual(result["message"],
+                         "You have successfully logged in!.")
+        self.assertTrue(result["access_token"])
+
+    def test_non_registered_user_login(self):
+        """Test non registered users cannot login."""
+        response = self.client().post('/api/v1/auth/signup', data=self.user_data)
```

Comment:
What is the essence of this line of code considering it has no impact on what you are testing for?

Reply:
It is used to test unregistered user. in that we first signup a user and then try to login with a different data set.